designs/images/snowmachineconfig-controller-container-components.puml

```diff
@@ -0,0 +1,24 @@
+@startuml snowmachineconfig-controller-container-components
+!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Component.puml
+
+title Component diagram for SnowMachineConfig controller
+
+Person(user, "EKS-A user", "Manages EKS-A clusters")
+Container(kubeApiServer, "Kubernetes API server", "Go", "Provides read/write access to the kubernetes API objects stored in the etcd server and allows to subscribe to events when they change")
+
+Container_Boundary(snowMachineConfigController, "Snow MachineConfig Controller") {
+    Component(snowMachineConfigReconciler, "SnowMachineConfig Reconciler", "K8s controller-runtime Reconciler", "Runs validations for SnowMAchineConfig that require external API calls (eg. Snow endpoint)")
+    Component(awsCli, "AWS API client", "awscli - Python")
+
+    Rel(snowMachineConfigReconciler, awsCli, "Uses")
+}
+
+Component_Ext(snowAPI, "Snow API endpoint")
+Rel(awsCli, snowAPI, "Makes API calls", "JSON/HTTPS")
```

Comment:
Just curious, only the cli works here and not the go sdk? Probably because it's not written for snow?

Reply:
awsCli, should have been `awsClient`. Updated and clarified on the component itself